tests/run-make/symlinked-extern/rmake.rs

```diff
@@ -16,7 +15,26 @@ use run_make_support::{cwd, rfs, rustc};
 fn main() {
     rustc().input("foo.rs").run();
     rfs::create_dir_all("other");
-    rfs::create_symlink("libfoo.rlib", "other");
+    #[cfg(unix)]
+    {
+        rfs::unix::symlink(cwd().join("libfoo.rlib"), cwd().join("other").join("libfoo.rlib"));
+    }
+    #[cfg(windows)]
+    {
+        rfs::windows::symlink_file(
```

Comment:
This seems quite cumbersome and error-prone. Can't we create more precise variants of the symlink functions (`symlink_file`, `symlink_dir` - or whatever other versions needed) and then let each runmake test decide which exact variant it needs, and handle the implementation difference itself in rfs?

Like this:
```rust
// runmake test
rfs::symlink_file("a", "b");
rfs::symlink_dir("x", "y");

// rfs
unix::symlink_file => symlink
unix::symlink_dir => symlink
windows::symlink_file => symlink_file
windows::symlink_dir => symlink_dir
```

Reply:
Yeah that seems reasonable. I'll change this tomorrow.

Reply:
(I lied I changed this today)